src/resolver.rs

```diff
@@ -102,13 +102,45 @@ struct DownloadError {
     version: VersionReq,
 }
 
+struct ProcessDependency<'a> {
+    name: PackageName,
+    dependency: Dependency,
+    is_root: bool,
+    lockfile: &'a Lockfile,
+    credentials: &'a Arc<Credentials>,
+    cache: &'a Cache,
+    preserve_mtime: bool,
+}
+
+struct ProcessLocalDependency<'a> {
+    name: PackageName,
+    dependency: LocalDependency,
+    #[allow(dead_code)]
```

Comment:
@mara-schulke refactoring to avoid the "too many function arguments" warning here, and it showed that this input is never read. I left it for you to deal with it as you see fit.